client/src/components/RepositoryFiles/index.tsx

```diff
@@ -12,34 +12,37 @@ import {
 
 type Props = {
   files: RepositoryFile[];
+  repositoryName: string;
   currentPath: string;
   onClick: (p: string, type: FileTreeFileType) => void;
 };
 
-const RepositoryFiles = ({ files, currentPath, onClick }: Props) => {
+const RepositoryFiles = ({
+  files,
+  currentPath,
+  onClick,
+  repositoryName,
+}: Props) => {
+  const parts = splitPathForBreadcrumbs(currentPath, (e, item, index, arr) => {
+    const path = breadcrumbsItemPath(arr, index, isWindowsPath(currentPath));
+    onClick(path, FileTreeFileType.DIR);
+  });
+  const [pathParts] = useState<PathParts[]>([
+    {
+      label: repositoryName,
+      onClick: () => onClick('/', FileTreeFileType.DIR),
+    },
+    ...parts,
+  ]);
```

Comment:
why do we need it as a state variable at all? Isn't it more logical to use useMemo here for pathParts